aviary/subsystems/propulsion/propulsion_mission.py

```diff
@@ -166,21 +166,15 @@ def setup(self):
             self.connect(engine.name + '.' + Dynamic.Mission.NOX_RATE,
                          'vectorize_performance.' + Dynamic.Mission.NOX_RATE + '_' + str(i))
 
-            try:
-                if engine.use_t4:
-                    self.connect(engine.name + '.' + Dynamic.Mission.TEMPERATURE_T4,
-                                 'vectorize_performance.' + Dynamic.Mission.TEMPERATURE_T4 + '_' + str(i))
-            except AttributeError:  # engine does not have flag
-                pass
-
-            try:
-                if engine.use_shp:
-                    self.connect(engine.name + '.' + Dynamic.Mission.SHAFT_POWER,
-                                 'vectorize_performance.' + Dynamic.Mission.SHAFT_POWER + '_' + str(i))
-                    self.connect(engine.name + '.' + Dynamic.Mission.SHAFT_POWER_CORRECTED,
-                                 'vectorize_performance.' + Dynamic.Mission.SHAFT_POWER_CORRECTED + '_' + str(i))
-            except AttributeError:  # engine does not have flag
-                pass
+            if getattr(engine, 'use_t4', False):
```

Comment:
Deja vu here, with a discussion we had in the openmdao dev channel on try vs check.  I agree that check is more appropriate and readable here.

Reply:
it also fixed an issue where `Dynamic.Mission.TEMPERATURE_T4` was being used in the past, but the variable in the hierarchy was `Dynamic.Mission.TEMPERATURE_ENGINE_T4`, which also caused an AttributeError that was caught by the try/except and hid an issue with the connections